src/lib/Strategy/Storage/ItemGroupListStrategy.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzRecommendationClient\Strategy\Storage;
+
+use EzSystems\EzRecommendationClient\Exception\UnsupportedGroupItemStrategy;
+use Ibexa\Contracts\Personalization\Criteria\CriteriaInterface;
+use Ibexa\Contracts\Personalization\Value\ItemGroupListInterface;
+
+final class ItemGroupListStrategy implements ItemGroupListStrategyInterface
+{
+    /** @var iterable<\EzSystems\EzRecommendationClient\Strategy\Storage\GroupItemStrategyInterface> */
+    private iterable $strategies;
+
+    /** @param iterable<\EzSystems\EzRecommendationClient\Strategy\Storage\GroupItemStrategyInterface> $strategies */
+    public function __construct(iterable $strategies)
+    {
+        $this->strategies = $strategies;
+    }
+
+    public function getGroupList(CriteriaInterface $criteria, string $groupBy): ItemGroupListInterface
+    {
+        foreach ($this->strategies as $strategy) {
+            if ($strategy->supports($groupBy)) {
```

Comment:
Since `Strategy` only depends for `supports` method on a simple string, you could turn it into an indexed iterator (https://symfony.com/doc/current/service_container/tags.html#tagged-services-with-index), which would mean you don't need to iterate on strategies to find the right one.